### PR TITLE
[SHELL32] Fix "Open With..." spamming registry MRU keys.

### PR DESCRIPTION
**Pull request by KyleKatarn31**
## Purpose

- When opening "Open With..." dialog for a given file format, multiple MRU items are appended to registry : 2 items written, even if dialog is closed "cancel".
- Expected behavior : 1 MRU item if execute is selected and 0 MRU if cancel is selected

JIRA issue: [CORE-16981](https://jira.reactos.org/browse/CORE-16981)

## Proposed changes

- MRU item write in registry only if "Execute" if selected (no MRU if "cancel")
- MRU item presence flag corectly set after writing MRU item to avoid multiple write action

### CODE REVIEW

**Comment by KyleKatarn31** [comment]
Does this comes from Wine or not ?

**Comment by binarymaster** [comment]
> Does this comes from Wine or not ?

It's our file, you can simply check it by looking at git history for file or directory: https://github.com/reactos/reactos/commits/master/dll/win32/shell32/COpenWithMenu.cpp

File: dll/win32/shell32/COpenWithMenu.cpp [hunk 1]
```diff
@@ -533,9 +533,7 @@ VOID COpenWithList::LoadFromProgIdKey(HKEY hKey, LPCWSTR pwszExt)
         if (pApp)
         {
             StringCbCopyW(pApp->wszCmd, sizeof(pApp->wszCmd), wszCmd);
-            SetRecommended(pApp);
-            if (!pApp->bMRUList)
-                AddAppToMRUList(pApp, pwszExt);
+            SetRecommended(pApp);
```
Comment (binarymaster) [inline dll/win32/shell32/COpenWithMenu.cpp:536]: Remove whitespace after `;`

Comment (KyleKatarn31) [reply dll/win32/shell32/COpenWithMenu.cpp]: OK !

Comment (binarymaster) [reply dll/win32/shell32/COpenWithMenu.cpp]: Thanks, now diff looks ok.

**Comment by KyleKatarn31** [comment]
Following today's discussion in Dev channel, I will not modify this code.
Either you merge as is, or you close and discard.

File: dll/win32/shell32/COpenWithMenu.cpp [hunk 2]
```diff
@@ -653,6 +651,9 @@ BOOL COpenWithList::AddAppToMRUList(SApp *pApp, LPCWSTR pwszFilename)
         /* Insert the entry */
         AddMRUStringW(hList, pApp->wszFilename);
 
+        /* Set MRU pressence */
```
Comment (HBelusca) [inline dll/win32/shell32/COpenWithMenu.cpp:654]:
```suggestion
        /* Set MRU presence */
```

**Comment by KyleKatarn31** [comment]
Thanks

**Pull request merged**